web/src/recent_topics_ui.js

```diff
@@ -728,6 +728,15 @@ function topic_sort(a, b) {
     return sort_comparator(topic_sort_key(a), topic_sort_key(b));
 }
 
+function unread_count(conversation_data) {
+    const message = message_store.get(conversation_data.last_msg_id);
+    return message_to_conversation_unread_count(message);
+}
+
+function unread_sort(a, b) {
+    return unread_count(a) > unread_count(b) ? 1 : -1;
```

Comment:
This is not a correct sorting function; returning `-1` when two values are equal can produce unfortunate results.

Reply:
Updated!